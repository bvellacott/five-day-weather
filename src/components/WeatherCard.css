.WeatherCard {
	background-color : #59f;
}

.WeatherCard:nth-child(even) {
	background-color : #48e;
}

.WeatherCard h1 {
	font-size: 4rem;
	display: inline;
}

.WeatherCard h2 {
	font-size: 3.5rem;
	display: inline;
}

.WeatherCard h3 {
	font-size: .85rem;
}

.WeatherCard img {
	height: 4rem;
	vertical-align: middle;
	display: inline;
}

.WeatherCard li {
	display: inline-block;
}
