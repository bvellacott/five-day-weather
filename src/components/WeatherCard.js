import React, { Component } from 'react';
import moment from 'moment';
import icons from '../lib/icons.js';
import './WeatherCard.css';

class Day extends Component {
  constructor(props) {
    super(props);
    this.m = moment(props.date*1000);
  }

  render() {
    return <h2>{this.m.format('dddd')}</h2>;
  }
}

class FromNow extends Component {
  constructor(props) {
    super(props);
    this.m = moment(props.date*1000);
  }

  render() {
    return <h3>{this.m.format('LT')}</h3>;
  }
}

function WeatherList(props) {
  return (
    <ul>
      { props.list.map((data) => <li key={data.dt} className="weather"><Weather data={data}/></li> ) }
    </ul>
  );
}

function Weather(props) {
  const date = props.data.dt;
  const weather = props.data.weather[0];
  return (
    <div>
      <FromNow date={date} />
      <img src={ icons[weather.icon]} alt={weather.description }/>
    </div>
  );
}

class WeatherCard extends Component {
  render() {
    return (
      <li className="WeatherCard">
        <Day date={this.props.list[0].dt}/>
        <WeatherList list={this.props.list}/>
      </li>
    );
  }
}

export default WeatherCard;
