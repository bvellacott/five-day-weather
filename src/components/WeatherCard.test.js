import React from 'react';
import WeatherCard from './WeatherCard';
import renderer from 'react-test-renderer';

test('Display data', () => {
  const component = renderer.create(
    <WeatherCard list={
  [{
    "dt": 1479438000,
    "weather": [{
      "description": "light rain",
      "icon": "10n"
    }],
  }, {
    "dt": 1479448800,
    "weather": [{
      "description": "clear sky",
      "icon": "01n"
    }],
  }]}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
