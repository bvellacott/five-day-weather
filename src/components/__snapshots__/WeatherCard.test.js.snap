exports[`test Display data 1`] = `
<div
  className="WeatherCard">
  <h2>
    Friday
  </h2>
  <ul>
    <li
      className="weather">
      <div>
        <h3>
          3:00 AM
        </h3>
        <img
          alt="light rain"
          src="test-file-stub" />
      </div>
    </li>
    <li
      className="weather">
      <div>
        <h3>
          6:00 AM
        </h3>
        <img
          alt="clear sky"
          src="test-file-stub" />
      </div>
    </li>
  </ul>
</div>
`;
